community/values/src/main/java/org/neo4j/values/storable/DateTimeValue.java

```diff
@@ -429,10 +435,42 @@ public boolean equals( Value other )
     }
 
     @Override
-    int unsafeCompareTo( Value otherValue )
+    int unsafeCompareTo( Value other )
+    {
+        ZonedDateTime that = ((DateTimeValue) other).value;
+        int cmp = Long.compare( value.toEpochSecond(), that.toEpochSecond() );
+        if ( cmp == 0 )
+        {
+            cmp = value.toLocalTime().getNano() - that.toLocalTime().getNano();
+            if ( cmp == 0 )
+            {
+                cmp = value.toLocalDateTime().compareTo( that.toLocalDateTime() );
+                if ( cmp == 0 )
+                {
+                    ZoneId thisZone = value.getZone();
+                    ZoneId thatZone = that.getZone();
+                    boolean thisIsOffset = thisZone instanceof ZoneOffset;
+                    boolean thatIsOffset = thatZone instanceof ZoneOffset;
+                    cmp = Boolean.compare( thatIsOffset, thisIsOffset ); // offsets before named zones
```

Comment:
I can't find this rule about offsets before named timezones in the CIP, is it documented somewhere?